### PR TITLE
Fix : Refactor hardcoded strings, values and colors in chooser_list_layout.xml file of collect.

### PR DESCRIPTION
**Pull request by supreme96**
While going through the code base for the collect app I noticed that very frequently in both java program files and xml layout files the strings, colors and values have been hardcoded instead of being defined separately and then referenced wherever required.
 This seems to be happening as contributors haven't really been following any specific coding style. Even the dimens.xml file doesn't exist in the repo. 
This is not a very good coding practice and has several drawbacks such as a lack of uniformity in the application UI, difficulty in maintaining the application as well as in troubles in localization and translations. 
More info on this can be obtained from these resources :
- http://stackoverflow.com/questions/8743349/hardcoded-string-row-three-should-use-string-resource
- https://nickbradbury.com/2013/12/08/android-tip-avoid-hard-coded-sizes/
- http://android4beginners.com/2013/06/appendix-b-everything-about-colors-in-android/ (see recommended option 6)

In this pull request(and some more on the way) I am beginning to address this issue by replacing these hardcoded resources by their reference to the exact same resources defined in the color.xml, strings.xml and dimens.xml. Doing this in no way affects the applications functionality and I have verified so by doing manual testing of the application. But doing so will allow us to create a fundamental base for implementing better coding style and standard for the Collect app's code base. 

In each commit the changes to the corresponding xml layout files have been also described. I intend to push changes to these files in a series of small PRs to make it easier to merge them.

Any suggestions and discussions are welcome :)

### CODE REVIEW

File: collect_app/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -19,4 +19,6 @@ the License.
     <drawable name="white">#ffffffff</drawable>
     <drawable name="black">#ff000000</drawable>
 
+    <color name="gainsboro">#DDDDDD</color>
```
Comment (nribeka) [inline collect_app/src/main/res/values/colors.xml]: #dcdcdc is gainsboro 🤣

File: collect_app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+
+    <!--Layout Dimensions-->
+    <dimen name="size_8">8dp</dimen>
+    <dimen name="size_4">4dp</dimen>
+    <dimen name="size_20">20dp</dimen>
+    <dimen name="size_250">250dp</dimen>
+
+
```
Comment (nribeka) [inline collect_app/src/main/res/values/dimens.xml]: So much blank spaces.

File: collect_app/src/main/res/values/colors.xml (hunk 1 above)
Comment (supreme96) [reply collect_app/src/main/res/values/colors.xml]: Yeah i know, but it's approximately gainsboro :p

File: collect_app/src/main/res/values/dimens.xml (hunk 2 above)
Comment (supreme96) [reply collect_app/src/main/res/values/dimens.xml]: Actually this PR is to begin the refactor. These blank spaces have been handled in subsequent commits.

**Review by lognaturel — CHANGES_REQUESTED** [review]
This is a good idea, @supreme96 and thanks for describing your process.

I think that it would be even better to use semantic naming that describes the purpose of the different resources rather than what they actually are. For example, "text_14" is a name that would have to be changed if we decided to change the size systematically. It also doesn't give any hints as to how different elements with text of size 14 are related. Better names might be something like "summary_text_size" or "medium_text_size". [Here](http://maintainablecss.com/chapters/semantics/) is a good article on naming. It's in the context of CSS but the same applies.

For colors, sometimes just giving the color a name is fine but same thing, if you can describe what it's for, that's even better. "accent_gray", "contrast_color", "spacer_color", things like that.

Even if you intend to submit several separate pull requests, @nribeka is right to ask that you clean up the extra spaces in `dimens.xml`. Each pull request should leave `master` in a clean state.

**Comment by supreme96** [comment]
@lognaturel some valid points that my view lacked. I'll get this working soon :)

**Comment by supreme96** [comment]
@lognaturel Requested changes in the PR have been made. Please review.

**Comment by lognaturel** [comment]
@supreme96 Can you please help me understand what the "_tv" suffix means?

I think this is a really great discussion and example but I'm wondering whether we should wait to make this part of the UI refresh since so much will be changing and standardizing.

File: collect_app/src/main/res/layout/chooser_list_layout.xml [hunk 3]
```diff
@@ -51,21 +51,23 @@ the License.
...
 
 	<!-- Side navigation drawer UI -->
 	<ListView
 		android:id="@+id/sortingMenu"
-		android:layout_width="250dp"
+		android:layout_width="@dimen/width_side_navigation_drawer"
 		android:layout_height="match_parent"
-		android:layout_gravity="end"
-		android:background="#DDDDDD"/>
+		android:layout_gravity="end|right"
```
Comment (lognaturel) [inline collect_app/src/main/res/layout/chooser_list_layout.xml]: This doesn't seem related to this pull request and it also makes the lint check for right-to-left languages fail! `end` will be `right` in the context of a left-to-right language and `left` in the context of a right-to-left language.

**Comment by supreme96** [comment]
@lognaturel The _tv suffix basically means a textview :)

I do realise that the code base is pending changes as the UI of the app will undergo a major redesign this summer, however it will be useful to work this out because it will ensure who ever works the redesign have a good example of improved and desired coding practices before major changes in the project are made.

**Comment by supreme96** [comment]
@lognaturel Fixed the `end|right` value for the layout_gravity attribute for navigation drawer in chooser_list_layout.xml. Please review.

**Comment by lognaturel** [comment]
Thanks so much for doing this @supreme96 and for pointing to so many great resources. I've gone back and forth on the right timing for this and ultimately I think it will be a good thing to do as part of a broader rethinking of the user interface. Ideally we will be able to come up with more standard dimensions that don't just refer to specific view elements but instead have some broader meaning. The discussion we've had here will be extremely helpful and I really look forward to putting these ideas into practice!

**Pull request closed without merging**